src/main/java/apoc/util/Util.java

```diff
@@ -210,40 +215,71 @@ public static Double toDouble(Object value) {
     }
 
     public static long toLong(Object value) {
-        if (value instanceof Number) return ((Number)value).longValue();
+        if (value instanceof Number) return ((Number) value).longValue();
         return Long.parseLong(value.toString());
     }
 
     public static URLConnection openUrlConnection(String url, Map<String, Object> headers) throws IOException {
+
         URL src = new URL(url);
-        URLConnection con = src.openConnection();
+        HttpURLConnection con = (HttpURLConnection) src.openConnection();
+        setConnectionParams(con, headers);
+        int code = con.getResponseCode();
+        return isRedirect(code) ? openUrlConnection(con.getHeaderField("Location"), headers) : con;
+    }
+
+    public static boolean isRedirect(int code) {
+        return code >= 300 && code < 400;
+    }
+
+    private static void setConnectionParams(HttpURLConnection con, Map<String, Object> headers) throws ProtocolException {
```

Comment:
Sure, we're trying to separate different protocols in dedicated pieces of code, in order to avoid test like `if (con instanceof HttpUrlConnection)` do X else do Y...
Let us review it.